src/Routes/GSheetImport.ts

```diff
@@ -14,13 +16,18 @@ export const upload = async (
   next: express.NextFunction
 ) => {
   const { spreadSheetID, sheetName } = req.body
+  const authorizedIDs = (SPREADSHEET_IDS_WHITELIST || "").split(",")
 
   if (!spreadSheetID || !sheetName) {
     return res
       .status(500)
       .send("A valid spreadSheetID and sheetName is required")
   }
 
+  if (!authorizedIDs.includes(spreadSheetID)) {
+    return res.status(500).send("A valid spreadSheetID is required")
```

Comment:
The casing of `sheet` could be made more consistent throughout the constant, code, and test. (Since the constant starts with `SPREADSHEET...`, I'd expect `spreadsheet` to be all lowercase.)